dynamo/src/main/scala/d4s/codecs/D4SCodec.scala

```diff
@@ -2,14 +2,33 @@ package d4s.codecs
 
 import software.amazon.awssdk.services.dynamodb.model.AttributeValue
 
-trait D4SCodec[A] extends D4SEncoder[A] with D4SDecoder[A]
+trait D4SCodec[A] extends D4SAttributeEncoder[A] with D4SAttributeDecoder[A] {
+  self =>
+
+  def encoder: D4SEncoder[A]
```

Comment:
Having them as fields isn't strictly neccessary, you could just make a constructor:
```scala
def fromPair[A](encoder: D4SDecoder[A], decoder: D4SDecoder[A]): D4SCodec[A] = new D4SCodec {
  override ...
}
```
and then `imap` is implementable with

```scala
D4SCodec.fromPair(contramap(from), map(to))
```

Reply:
Don't we want to give a user the opportunity to work with decoder derived with codec?

Reply:
? Didn't get it.
I mean if you also return back `D4SCodec[A] extends D4SEncoder[A] with D4SDecoder[A]`